src/librustdoc/clean/utils.rs

```diff
@@ -129,24 +122,68 @@ pub(crate) fn ty_args_to_args<'tcx>(
                 return None;
             }
 
-            if !elision_has_failed_once_before
+            if !*elision_has_failed_once_before
                 && let Some(default) = params[index].default_value(cx.tcx)
             {
+                match default.skip_binder().unpack() {
+                    GenericArgKind::Const(_) => {}
+                    GenericArgKind::Type(_) | GenericArgKind::Lifetime(_) => {
```

Comment:
This is treating symptoms not the root cause, we shouldn't have a mismatch between `generics` and `ty_args` in the first place. The root cause is an off-by-one introduced by me since I forgot to handle trait object types correctly whose generics args don't contain a `Self` type.

I'm gonna submit an alternative fix shortly.

Your PR also doesn't fix (you'd need to update the outer `GenericArgKind::Type` branch, too):

```rs
pub trait Trait<const K: u32 = 0, T = ()> {}
pub type Alias<const K: u32> = dyn Trait<K>;
```

```rs
pub use dependency::*;
```